cql3/statements/truncate_statement.cc

```diff
@@ -8,32 +8,71 @@
  * SPDX-License-Identifier: (AGPL-3.0-or-later and Apache-2.0)
  */
 
+#include "cql3/statements/raw/truncate_statement.hh"
 #include "cql3/statements/truncate_statement.hh"
 #include "cql3/statements/prepared_statement.hh"
 #include "cql3/cql_statement.hh"
 #include "data_dictionary/data_dictionary.hh"
 #include "cql3/query_processor.hh"
 #include "service/storage_proxy.hh"
 #include <optional>
+#include "validation.hh"
 
 namespace cql3 {
 
 namespace statements {
 
-truncate_statement::truncate_statement(cf_name name)
-    : cf_statement{std::move(name)}
-    , cql_statement_no_metadata(&timeout_config::truncate_timeout)
+namespace raw {
+
+truncate_statement::truncate_statement(cf_name name, std::unique_ptr<attributes::raw> attrs)
+        : cf_statement(std::move(name))
+        , _attrs(std::move(attrs))
 {
+    // Validate the attributes.
+    // Currently, TRUNCATE supports only USING TIMEOUT
+    if (_attrs->time_to_live) {
+        throw exceptions::invalid_request_exception("TRUNCATE does not support USING TTL");
+    }
+    if (_attrs->timestamp) {
+        throw exceptions::invalid_request_exception("TRUNCATE does not support USING TIMESTAMP");
+    }
```

Comment:
If we later add more of these USING options, this won't check them. Would have been nice if we could verify that **only** the timeout option is set. I'm not sure how. One way this could maybe be done is to create an attr object with *only* the timeout and then check if the two objects are equal (requires an operator== that works properly).

But maybe this code is good enough, and we can improve it later if we add more options.

Reply:
I think it's better to restrict it in the grammar.

Reply:
ok. Will change for `selectStatement` too then.

Reply:
Ref: https://github.com/scylladb/scylladb/blob/e5e75ba43c79a575c44f9b0824374b2be0798b72/cql3/statements/select_statement.cc#L1573-L1580

Reply:
addressed in v3